test/couch_httpd_headers_tests.erl

```diff
@@ -0,0 +1,55 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_httpd_headers_tests).
+
+-include_lib("couch/include/couch_eunit.hrl").
+
+add_header_if_needed_test() ->
+    Cases = [
+        % should add to empty headers list
+        {[],                           {"K1", "V1"}, [{"K1", "V1"}]},
+        % should add to non-empty header_list
+        {[{"K1", "V1"}],               {"K2", "V2"}, [{"K1", "V1"}, {"K2", "V2"}]},
+         % should not add or change when already there
+        {[{"K1", "V1"}],               {"K1", "V2"}, [{"K1", "V1"}]},
+         % should not add or change when already there
+        {[{"K1", "V1"}, {"K2", "V2"}], {"K1", "V2"}, [{"K1", "V1"}, {"K2", "V2"}]},
+        % should add to a bit longer list
+        {[{"K1", "V1"}, {"K2", "V2"}], {"K3", "V3"}, [{"K1", "V1"}, {"K2", "V2"}, {"K3", "V3"}]} 
+    ],
+    lists:map(fun({InitialHeaders, HeaderToAdd, ProperResult}) ->
```

Comment:
Here you need to use [tests generation](http://www.erlang.org/doc/apps/eunit/chapter.html#id61504) - scroll down to "Writing test generating functions".